### PR TITLE
JAMES-2992 JMAP fast view read profile

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/MessageProperties.java [hunk 1]
```diff
@@ -217,7 +217,8 @@ private static boolean outputProperty(MessageProperty p) {
     public enum ReadProfile {
         Metadata(0, FetchGroup.MINIMAL),
         Header(1, FetchGroup.HEADERS),
-        Full(2, FetchGroup.FULL_CONTENT);
+        Fast(2, FetchGroup.HEADERS),
```
Comment (chibenwa) [inline server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/MessageProperties.java]: Given our new design, as the MessageViewFactory end up doing the read, I am not sure it makes sense any longer to be sprecifying the FetchGroup here. 

Technically that is wrong: Fast FetchGroup can be either Headers or Full, depending on "the projection content..."

**Comment by chibenwa** [comment]
```
[0244a2cfe64d55e9bd404dcfa46299563ad791c7] [ERROR] src/main/java/org/apache/james/jmap/draft/model/message/view/MessageMetadataViewFactory.java:[28,8] (imports) UnusedImports: Unused import - org.apache.james.jmap.draft.model.MessageProperties.ReadProfile.
[0244a2cfe64d55e9bd404dcfa46299563ad791c7] [ERROR] src/main/java/org/apache/james/jmap/draft/model/message/view/MessageHeaderViewFactory.java:[30,8] (imports) UnusedImports: Unused import - org.apache.james.jmap.draft.model.MessageProperties.
[0244a2cfe64d55e9bd404dcfa46299563ad791c7] [ERROR] src/main/java/org/apache/james/jmap/draft/model/message/view/MessageFullViewFactory.java:[38,8] (imports) UnusedImports: Unused import - org.apache.james.jmap.draft.model.MessageProperties.
```

**Comment by trantienduchn** [comment]
was :green_apple:

File: server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/MessageProperties.java (hunk 1 above)
Comment (mbaechler) [inline server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/MessageProperties.java]: I don't understand the mapping between ReadProfile and FetchGroup

Comment (trantienduchn) [reply server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/MessageProperties.java]: my fault, the FetchGroup was removed by Benoit BTW, I forgot to fetch

**Comment by trantienduchn** [comment]
```
[72afcced8da0d1f2a50d946e212fcb550a7f6b4b] [ERROR] Errors: 
[72afcced8da0d1f2a50d946e212fcb550a7f6b4b] [ERROR]   MemorySpamAssassinContractTest.deletingSpamMessageShouldNotImpactSpamAssassinState » ConditionTimeout
[72afcced8da0d1f2a50d946e212fcb550a7f6b4b] [ERROR]   MemorySpamAssassinContractTest.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam » ConditionTimeout
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**